apache/trafficcontrol
# TP: allows topology-based delivery services to be assigned to ORG servers

...and fixes topology name validation in TP.

## What does this PR (Pull Request) do?

- [x] This PR fixes #5197 and fixes #5161 

## Which Traffic Control components are affected by this PR?

- Traffic Portal

## What is the best way to verify this PR?
- Verify UI tests pass
- Create a topology. Ensure that the name can only consist of alphanumeric, dash, or underscore characters.
- Assign the topology to a delivery service. Note the xml-id of this topology-based delivery service. Note the CDN of this DS.
- Navigate to an EDGE server in the same CDN at https://tp.domain.tld/#!/servers/{EDGE-ID}/delivery-services and try to assign your topology-based delivery service to the EDGE server. It should NOT be possible. All topology-based delivery services are disabled for selection.
- Navigate to an ORG server in the same CDN at https://tp.domain.tld/#!/servers/{ORG-ID}/delivery-services and try to assign your topology-based delivery service to the ORG server. It should be possible.

## If this is a bug fix, what versions of Traffic Control are affected?
- master
- 5.0.0 RC1

## The following criteria are ALL met by this PR



## Additional Information
